include/xtensor/xrepeat.hpp

```diff
@@ -0,0 +1,442 @@
+        using pointer = typename xexpression_type::pointer;
+        using const_pointer = typename xexpression_type::const_pointer;
+        using difference_type = typename xexpression_type::difference_type;
+
+        using iterable_type = xiterable<xrepeat<CT>>;
+        using stepper = typename iterable_type::stepper;
+        using const_stepper = typename iterable_type::stepper;
+
+        template<class CTA>
+        explicit xrepeat(CTA&& e,
+            std::vector<std::ptrdiff_t>&& repeats,
+            std::vector<std::ptrdiff_t>&& axes);
+
+        template <class... Args>
+        reference operator()(Args... args);
+
+        template <class... Args>
+        const_reference operator()(Args... args) const;
+
+        const shape_type& shape() const noexcept;
+
+        template <class It>
+        reference element(It first, It last);
+
+        template <class It>
+        const_reference element(It first, It last) const;
+
+        stepper stepper_begin(const shape_type& s) const;
+
+        stepper stepper_end(const shape_type& s, const layout_type l) const;
+
+    private:
+        CT m_e;
+        std::vector<std::ptrdiff_t> m_repeat_lookup;
+        shape_type m_shape;
+
+        reference access();
+
+        template <class Arg, class... Args>
+        reference access(Arg arg, Args... args);
+
+        const_reference access() const;
+
+        template <class Arg, class... Args>
+        const_reference access(Arg arg, Args... args) const;
+
+        template<typename std::decay_t<CT>::size_type I>
+        struct access_impl
+        {
+
+            template<class Arg, class... Args>
+            inline const_reference operator()(stepper& s, Arg arg, Args... args) const
+            {
+                s.step(I, arg);
+                const access_impl<I + 1> access_next_dimension;
+                return access_next_dimension(s, args...);
+            }
+
+            inline const_reference operator()(stepper& s) const
+            {
+                return *s;
+            }
+
+            template<class Arg, class... Args>
+            inline reference operator()(stepper& s, Arg arg, Args... args)
+            {
+                s.step(I, arg);
+                access_impl<I + 1> access_next_dimension;
+                return access_next_dimension(s, args...);
+            }
+
+            inline reference operator()(stepper& s)
+            {
+                return *s;
+            }
+        };
+
+        access_impl<0> m_access_impl;
+    };
+
+    template <class CT>
+    template <class CTA>
+    xrepeat<CT>::xrepeat(CTA&& e, std::vector<std::ptrdiff_t>&& repeats, std::vector<std::ptrdiff_t>&& axes)
+        : m_e(std::forward<CTA>(e))
+        , m_repeat_lookup(e.dimension())
+        , m_shape(e.shape())
+    {
+        std::fill(m_repeat_lookup.begin(), m_repeat_lookup.end(), 1);
+        for (auto index = 0; index < axes.size(); ++index)
+        {
+            const auto axis = axes.at(index);
```

Comment:
unchecked access operator should be prefered for performance consideration (although it might not make a big difference in the constructor, it could in other parts of the code). If bound checking is somehow wanted, we can add some assertions that would be enabled explicitly by the user.

Reply:
Sorry, wasn't aware of this bound check. Thanks for your feedback.